utils/results/conversion/sarifparser/sarifparser.go

```diff
@@ -127,89 +171,117 @@ func (sc *CmdResultsSarifConverter) validateBeforeParse() (err error) {
 	if sc.current == nil {
 		return results.ErrResetConvertor
 	}
-	if (sc.hasViolationContext || sc.includeVulnerabilities) && sc.scaCurrentRun == nil {
+	if sc.currentTarget == nil {
 		return results.ErrNoTargetConvertor
 	}
 	return
 }
 
-func (sc *CmdResultsSarifConverter) ParseViolations(target results.ScanTarget, scanResponse services.ScanResponse, applicabilityRuns ...*sarif.Run) (err error) {
-	if err = sc.validateBeforeParse(); err != nil || sc.scaCurrentRun == nil {
+func (sc *CmdResultsSarifConverter) ParseScaIssues(target results.ScanTarget, violations bool, scaResponse results.ScanResult[services.ScanResponse], applicableScan ...results.ScanResult[[]*sarif.Run]) (err error) {
+	if violations {
+		if err = sc.parseScaViolations(target, scaResponse, applicableScan...); err != nil {
+			return
+		}
+		return
```

Comment:
Question: I remomber we said that we can have both vulnerabilities and violations together. should we take out this return or is it not possible in here?

Reply:
same logic for all parser, it will be parsed in different calls